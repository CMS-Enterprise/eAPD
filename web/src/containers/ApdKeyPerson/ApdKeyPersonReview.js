import PropTypes from 'prop-types';
import React, { Fragment, useMemo } from 'react';
import Dollars from '../../components/Dollars';
import Review from '../../components/Review';

const ApdStateKeyPerson = ({
  expand,
  index,
  item: { costs, email, hasCosts, name, percentTime, position },
  onDeleteClick
}) => {
  const primary = index === 0;

  const totalCost = useMemo(
    () => Object.values(costs).reduce((sum, value) => sum + +value, 0),
    [costs]
  );

  return (
    <Fragment>
      <div className="visibility--screen">
        <Review
          heading={`${index + 1}. ${name}`}
          onDeleteClick={onDeleteClick}
          onEditClick={expand}
        >
          <ul className="ds-c-list--bare">
            <li>{position}</li>
            <li>
              Total cost: <Dollars>{hasCosts ? totalCost : 0}</Dollars>
            </li>
            {primary ? <li>Primary APD Point of Contact</li> : null}
          </ul>
        </Review>
      </div>
      <div className="visibility--print">
        <Review heading={`${index + 1}. ${name}`}>
          <ul className="ds-c-list--bare">
            <li>{email}</li>
            <li>{position}</li>
            <li>Time commitment to project: {percentTime}%</li>
            <li>
              Total cost: <Dollars>{hasCosts ? totalCost : 0}</Dollars>
            </li>
            {primary ? <li>Primary APD Point of Contact</li> : null}
          </ul>
        </Review>
      </div>
    </Fragment>
  );
};

ApdStateKeyPerson.propTypes = {
  expand: PropTypes.func.isRequired,
  index: PropTypes.number.isRequired,
  item: PropTypes.shape({
    costs: PropTypes.object.isRequired,
    email: PropTypes.string.isRequired,
    hasCosts: PropTypes.bool.isRequired,
    name: PropTypes.string.isRequired,
    percentTime: PropTypes.number.isRequired,
    position: PropTypes.string.isRequired
  }).isRequired,
  onDeleteClick: PropTypes.func
};

ApdStateKeyPerson.defaultProps = {
  onDeleteClick: null
};

export default ApdStateKeyPerson;
