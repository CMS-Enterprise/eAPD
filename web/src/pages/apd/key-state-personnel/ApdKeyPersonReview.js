import PropTypes from 'prop-types';
import React, { Fragment, useMemo } from 'react';
import Dollars from '../../../components/Dollars';
import Review from '../../../components/Review';

const ApdStateKeyPerson = ({
  expand,
  index,
  item: { costs, email, hasCosts, name, fte, position },
  onDeleteClick
}) => {
  const primary = index === 0;
  let displayName = name;
  if (name === '') {
    displayName = primary
      ? 'Primary Point of Contact name not specified'
      : 'Key Personnel name not specified';
  }

  console.log({hasCosts})

  const costByYear = useMemo(
    () => (
      <div className="ds-u-margin-top--2">
        {hasCosts === true ? (
          Object.keys(costs).map(year => (
            <div key={year}>
              <strong>FFY {year} Cost: </strong>
              <Dollars>{costs[year]}</Dollars> | <strong>FTE: </strong>
              {fte[year]} | <strong>Total: </strong>
              <Dollars>{costs[year] * fte[year]}</Dollars>
            </div>
          ))
        ) : (
          <div>
            <strong>Total cost:</strong> <Dollars>{0}</Dollars>
          </div>
        )}
      </div>
    ),
    [hasCosts, costs, fte]
  );

  return (
    <Fragment>
      <div className="visibility--screen">
        <Review
          heading={`${index + 1}. ${displayName}`}
          headingLevel="4"
          onDeleteClick={primary ? null : onDeleteClick}
          onEditClick={expand}
          ariaLabel={displayName}
          objType="Key Personnel"
        >
          <ul className="ds-c-list--bare">
            {primary ? <li>Primary APD Point of Contact</li> : null}
            <li>{position || 'Role not specified'}</li>
          </ul>
          {costByYear}
        </Review>
      </div>
      <div className="visibility--print">
        <Review heading={`${index + 1}. ${displayName}`}>
          <ul className="ds-c-list--bare">
            {primary ? <li>Primary APD Point of Contact</li> : null}
            <li>{position || 'Role not specified'}</li>
            <li>{email}</li>
          </ul>
          {costByYear}
        </Review>
      </div>
    </Fragment>
  );
};

ApdStateKeyPerson.propTypes = {
  expand: PropTypes.func.isRequired,
  index: PropTypes.number.isRequired,
  item: PropTypes.shape({
    costs: PropTypes.object.isRequired,
    email: PropTypes.string.isRequired,
    hasCosts: PropTypes.bool.isRequired,
    name: PropTypes.string.isRequired,
    fte: PropTypes.object.isRequired,
    position: PropTypes.string.isRequired
  }).isRequired,
  onDeleteClick: PropTypes.func
};

ApdStateKeyPerson.defaultProps = {
  onDeleteClick: null
};

export default ApdStateKeyPerson;
