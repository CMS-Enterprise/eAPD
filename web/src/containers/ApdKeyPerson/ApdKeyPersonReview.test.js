import { shallow } from 'enzyme';
import React from 'react';

import ApdKeyPersonReview from './ApdKeyPersonReview';

describe('the ApdKeyPersonReview component', () => {
  const props = {
    expand: jest.fn(),
    index: 12,
    item: {
      costs: {
        '1992': 100,
        '1993': 300
      },
      email: 'email address',
      hasCosts: true,
      key: 'person key',
      name: 'Bob the Builder',
      percentTime: {
        '1992': 0.32,
        '1993': 0.84
      },
      position: 'The Builder'
    },
    onDeleteClick: jest.fn(),
    years: ['1992', '1993']
  };

  test('renders correctly', () => {
    const component = shallow(<ApdKeyPersonReview {...props} />);
    expect(component).toMatchSnapshot();

    expect(
      component
        .find('StandardReview')
        .first()
        .prop('onEditClick')
    ).toEqual(props.expand);

    expect(
      component
        .find('StandardReview')
        .first()
        .prop('onDeleteClick')
    ).toEqual(props.onDeleteClick);
  });
});
