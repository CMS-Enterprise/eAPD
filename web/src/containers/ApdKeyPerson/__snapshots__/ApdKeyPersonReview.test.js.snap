// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ApdKeyPersonReview component renders correctly 1`] = `
<Fragment>
  <div
    className="visibility--screen"
  >
    <StandardReview
      editHref={null}
      heading="13. Bob the Builder"
      headingLevel={4}
      onDeleteClick={[MockFunction]}
      onEditClick={[MockFunction]}
    >
      <ul
        className="ds-c-list--bare"
      >
        <li>
          The Builder
        </li>
        <li>
          <strong>
            Total cost:
          </strong>
           
          <Dollars
            long={false}
          >
            400
          </Dollars>
        </li>
      </ul>
    </StandardReview>
  </div>
  <div
    className="visibility--print"
  >
    <StandardReview
      editHref={null}
      heading="13. Bob the Builder"
      onDeleteClick={null}
      onEditClick={null}
    >
      <ul
        className="ds-c-list--bare"
      >
        <li>
          The Builder
        </li>
        <li>
          email address
        </li>
        <li>
          <strong>
            Time commitment to project:
          </strong>
           
          32
          %
        </li>
        <li>
          <strong>
            Total cost:
          </strong>
           
          <Dollars
            long={false}
          >
            400
          </Dollars>
        </li>
      </ul>
    </StandardReview>
  </div>
</Fragment>
`;
